src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java

```diff
@@ -258,19 +260,42 @@ public String search(String queryString, JsonNode selectedFacetsAndFields, Strin
         return response.toString();
     }
 
-
     @Override
     public InputStreamResource getStudyAsStream(String accession, String relativePath, boolean anonymise, Constants.File.StorageMode storageMode)
             throws IOException {
+        return getStudyAsStream(accession, relativePath, anonymise, storageMode, true);
+    }
+
+    @Override
+    public InputStreamResource getStudyAsStream(String accession, String relativePath, boolean anonymise, Constants.File.StorageMode storageMode, boolean fillPagetabIndex)
+            throws IOException {
 
         InputStream inputStream = null;
-        switch (storageMode) {
-            case FIRE:
-                inputStream = fireService.cloneFireS3ObjectStream(relativePath + "/" + accession + ".json");
-                break;
-            default:
-                inputStream = new FileInputStream(Paths.get(indexConfig.getFileRootDir(), relativePath, accession + ".json").toFile());
+        Exception codeIOException = null;
+        if (fillPagetabIndex) {
+            try {
+                inputStream = getPagetabFromPagetabIndex(accession);
+            } catch (Exception ex) {
+                codeIOException = ex;
+                logger.error("problem in searching pagetab index", ex);
+            }
+        }
+        try {
+            if (inputStream == null) {
```

Comment:
We should push the document in the index when we load it from NFS/FIRE

Reply:
So do you agree to remove saving pagetab in indexone service and just keep this cache gradually filling service?
I modified it some how that add pagetab from mongodb dump and indexone message in the same mechanism. Is this fine? (as you said the mongodb pagetab has some missing values)

Reply:
MongoDB documents are dumps of the internal model and do not follow the public PageTab specs. This new index should be for public PageTab only (available on FTP/Aspera/Globus etc) so we should probably add the public version to the index after loading it from NFS/FIRE.